File: cloud-functions/submitTrade/trade/tradeExecution.js
const createSession = require('./sessionFactory.js');
const CustomError = require('../utils/error.js');

async function tradeExecution(apiKey, apiSecret, apiPassphrase, method, data) {
    
    try {


        // THIS IS ALL FOR TESTING PURPOSES CODE NOT FINISHED
        // THIS IS ALL FOR TESTING PURPOSES CODE NOT FINISHED
        // THIS IS ALL FOR TESTING PURPOSES CODE NOT FINISHED
        // THIS IS ALL FOR TESTING PURPOSES CODE NOT FINISHED
        let exchange;
        let session;
        let tradeExecution;
        switch(method) {
            case 'bulkOrder':
                exchange = data.trader_exchange;
                session = createSession(exchange, apiKey, apiSecret, apiPassphrase);
                tradeExecution = await session.bulkOrder(data);
                break;
            case 'sendSl':
                exchange = data.trader_exchange;
                session = createSession(exchange, apiKey, apiSecret, apiPassphrase);
                tradeExecution = await session.sendSl(data);
                break;
            case 'replaceSL':
                exchange = data.trader_exchange;
                session = createSession(exchange, apiKey, apiSecret, apiPassphrase);
                tradeExecution = await session.replaceSL(data);
                break;
            case 'bulkTP':
                exchange = data.trader_exchange;
                session = createSession(exchange, apiKey, apiSecret, apiPassphrase);
                tradeExecution = await session.bulkTP(data);
                break;
            case 'cancelOrder':
                exchange = data.trader_exchange;
                session = createSession(exchange, apiKey, apiSecret, apiPassphrase);
                tradeExecution = await session.cancelOrder(data);
                break;
            case 'cancelAllTPs':
                exchange = data.trader_exchange;
                session = createSession(exchange, apiKey, apiSecret, apiPassphrase);
                tradeExecution = await session.cancelAllTPs(data);
                break;
            case 'cancelAllOrders':
                exchange = data.trader_exchange;
                session = createSession(exchange, apiKey, apiSecret, apiPassphrase);
                tradeExecution = await session.cancelAllOrders(data);
                break;
            case 'partialClose':
                exchange = data.trader_exchange;
                session = createSession(exchange, apiKey, apiSecret, apiPassphrase);
                tradeExecution = await session.partialClose(data);
                break;
        }

        return tradeExecution
    } catch(error) {
        throw new CustomError({
            message: `Error handling the trade: ${error.message}`,
            status: 500,
            source: 'tradeExecution',
        });
    }
}

module.exports = tradeExecution;

File: cloud-functions/submitTrade-py/trade/trade_execution.py
from .session_factory import create_session
import logging

logger = logging.getLogger(__name__)


async def trade_execution(api_key, api_secret, api_passphrase, method, data):
    try:

        trade_execution_result = None
        if method == 'bulk_order':
            exchange = data['trader_exchange']
            session = create_session(exchange, api_key, api_secret, api_passphrase)
            trade_execution_result = await session.bulk_order(data)
        elif method == 'send_sl':
            exchange = data['trader_exchange']
            session = create_session(exchange, api_key, api_secret, api_passphrase)
            trade_execution_result = await session.send_sl(data)
        elif method == 'replace_sl':
            exchange = data['trader_exchange']
            session = create_session(exchange, api_key, api_secret, api_passphrase)
            trade_execution_result = await session.replace_sl(data)
        elif method == 'bulk_tp':
            exchange = data['trader_exchange']
            session = create_session(exchange, api_key, api_secret, api_passphrase)
            trade_execution_result = await session.bulk_tp(data)
        elif method == 'cancel_order':
            exchange = data['trader_exchange']
            session = create_session(exchange, api_key, api_secret, api_passphrase)
            trade_execution_result = await session.cancel_order(data)
        elif method == 'cancel_all_tps':
            exchange = data['trader_exchange']
            session = create_session(exchange, api_key, api_secret, api_passphrase)
            trade_execution_result = await session.cancel_all_tps(data)
        elif method == 'cancel_all_orders':
            exchange = data['trader_exchange']
            session = create_session(exchange, api_key, api_secret, api_passphrase)
            trade_execution_result = await session.cancel_all_orders(data)
        elif method == 'partial_close':
            exchange = data['trader_exchange']
            session = create_session(exchange, api_key, api_secret, api_passphrase)
            trade_execution_result = await session.partial_close(data)

        return trade_execution_result

    except Exception as e:
        logger.error("An error occurred: %s", e, exc_info=True)
